Update app and runtime

wjt opened the pull request:
Fixes #6 inasmuch as it builds.
Fixes #7 except a newer runtime.

wjt committed:
Add shared-modules submodule

This will be needed for intltool in a future commit.

wjt committed:
Normalise manifest formatting

A future commit will add flatpak-external-data-checker annotations, and
fedc will reformat this file slightly.

wjt committed:
Add flatpak-external-data-checker metadata

wjt committed:
Update to GNOME 42 runtime & gbrainy 2.4.4

This fails to build for reasons fixed in a future commit.

hfiguiere commented on org.gnome.gbrainy.json:
```diff
@@ -3,56 +3,68 @@
     "branch": "stable",
     "runtime": "org.gnome.Platform",
-    "runtime-version": "3.36",
+    "runtime-version": "41",
```
`42` is the latest.

wjt replied:
Good catcH!

hfiguiere commented on org.gnome.gbrainy.json:
```diff
@@ -3,56 +3,68 @@
             "name": "gtk-sharp3",
-            "config-opts": [
-                           ],
+            "config-opts": [],
```
you could as well remove it.

```suggestion
```

hfiguiere commented on org.gnome.gbrainy.json:
```diff
@@ -3,56 +3,68 @@
+        "append-ld-library-path": "/usr/lib/sdk/mono6/lib",
         "env": {
-            "PKG_CONFIG_PATH": "/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:/usr/lib/sdk/mono5/lib/pkgconfig"
+            "PKG_CONFIG_PATH": "/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:/usr/lib/sdk/mono6/lib/pkgconfig"
```
`"append-pkg-config-path": "/usr/lib/sdk/mono6/lib/pkgconfig"` is probably better here.

hfiguiere replied:
just to clarify it would be one level up out of the `env` object.

wjt committed:
Install mono to /app before building gbrainy

This fixes the following build error:

    Making all in tools
    make[2]: Entering directory '/run/build/gbrainy/tools'
    /usr/lib/sdk/mono6/bin/mcs -target:winexe -out:GameXmlToGetString.exe  ./GameXmlToGetString.cs -r:../src/gbrainy.Core.dll -r:Mono.Posix
    export MONO_PATH=../src && /usr/lib/sdk/mono6/bin/mono GameXmlToGetString.exe .

    Unhandled Exception:
    System.TypeLoadException: Could not set up parent class, due to: Could not load type of field 'gbrainy.Core.Main.Game:default_color' (0) due to: Could not load file or assembly 'cairo-sharp, Version=1.10.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756' or one of its dependencies. assembly:/run/build/gbrainy/src/gbrainy.Core.dll type:Game member:(null)
      at GameXmlToGetString.Main (System.String[] args) [0x00061] in <2326c82eb00343b4a364a997caa15399>:0
    [ERROR] FATAL UNHANDLED EXCEPTION: System.TypeLoadException: Could not set up parent class, due to: Could not load type of field 'gbrainy.Core.Main.Game:default_color' (0) due to: Could not load file or assembly 'cairo-sharp, Version=1.10.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756' or one of its dependencies. assembly:/run/build/gbrainy/src/gbrainy.Core.dll type:Game member:(null)
      at GameXmlToGetString.Main (System.String[] args) [0x00061] in <2326c82eb00343b4a364a997caa15399>:0

I think the problem is that the gtk-sharp build step is adding the
.dll to (a subdirectory of) /app, but because Mono is being run from (a
subdirectory of) /usr that path is not in Mono's search path.

wjt committed:
Don't specify branch: stable in manifest

Flathub passes this on the CLI; and it is desirable for local builds to
use a clearly different branch.

wjt committed:
Build and install libcanberra

gbrainy uses this to play sounds. Without it you see the following on
stderr when gbrainy tries to play a sound:

    Unix.PlaySound. Error System.DllNotFoundException: libcanberra-gtk3.so.0 assembly:<unknown assembly> type:<unknown type> member:(null)
      at (wrapper managed-to-native) gbrainy.Core.Platform.Unix.ca_gtk_play_for_widget(intptr,uint,string,string,string,string,intptr)
      at gbrainy.Core.Platform.Unix.PlaySound (System.IntPtr widget, System.String filename) [0x00000] in <c1cbe592d22c4597a90c664ab477780f>:0

wjt commented:
Cracked it!

wjt commented:
Test build works on my machine™

jordimas commented:
Thanks, really appreciated
